event_scheduler/event_scheduler.py

```diff
@@ -48,10 +61,9 @@ def cancel(self, event):
         """
         with self._scheduler_status_lock:
             if self._scheduler_status != self.SchedulerStatus.RUNNING:
-                # TODO: Add error message
                 return -1
-        super().cancel(event)
-        return 0
+            super().cancel(event)
```

Comment:
The event can be cancelled even in stopping mode as long as the event hasn't been popped from the queue. It's good you brought attention here. There should be a notify here since the state of the queue has changed (it's actually possible that we cancel an event that's at the front of the queue). We should recalculate the new timer right away. I'm pretty sure it would still work the way it is right now since when the timer expires, it will recalculate the for the new event at the front of the queue anyways (but this is kinda sloppy).

Reply:
hmm yea it seems like just an edge case that might not be worth he time